ceph-releases/luminous/ubuntu/16.04/daemon/exit.sh

```diff
@@ -21,16 +21,25 @@ function umount_ceph_mnt {
   done
 }
 
-function handler {
-  while true; do
-    tail -f /dev/null & wait ${!}
-  done
-}
-
 function graceful_stop {
-  handler
-  kill -SIGTERM 1
-  wait 1
+  echo "Sending SIGTERM to PID $child_for_exec"
+  kill -SIGTERM "$child_for_exec"
+  wait "$child_for_exec"
+
   umount_ceph_mnt
-  exit 143; # 128 + 15 -- SIGTERM
+  exit 143 # 128 + 15 -- SIGTERM
+}
+
+child_for_exec=1
+
+function exec {
+  # This function overrides the built-in exec() call
+  # It starts the process in background to catch ERR but
+  # as per docker requirement, forward the SIGTERM to it.
+  trap graceful_stop SIGTERM
```

Comment:
You lack of the set -E to forward  the trap handler to all functions. Ideally to set at the beginning of the program.

Reply:
Why don't use manager ERR ? If the exec'd program crash you have to manage the cleanup isn't it ?